feat: adding ‘retainSelection’ to TabBar

archaica opened the pull request:
## Description

New 'retainSelection' property allows selected tab to remain ‘selected’ in mode even when the TabBar loses focus. That way, if the content is not using the 'TabContent' area in TabBar, we can maintain the same sort of relationship between 'selected' content and retain the visual identity of the selected tab.

## References

## Testing

I added a test to verify this change.

## Automation

## Checklist

- [ ] all commented code has been removed
- [ ] any new console issues have been resolved
- [ ] code linter and formatter has been run
- [ ] test coverage meets repo requirements
- [ ] PR name matches the expected semantic-commit syntax

benmunson committed:
feature: adding ‘retainSelection’ to TabBar

Property allows selected tab to remain ‘selected’ in mode even when the TabBar loses focus.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/rdkcentral/Lightning-UI-Components?pullRequest=488) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/rdkcentral/Lightning-UI-Components?pullRequest=488) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: archaica<br/>:x: benmunson<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/rdkcentral/Lightning-UI-Components?pullRequest=488) it.</sub>

svc-lightning-ui-components commented:
Test Execution Passed.

svc-lightning-ui-components commented:
Test Execution Passed.

joshhowenstine reviewed (approved):
Looks great, thanks for contributing!

anthony9187 commented on packages/@lightningjs/ui-components/src/components/TabBar/TabBar.test.js:
```diff
@@ -354,4 +354,16 @@ describe('TabBar', () => {
+
+    tabBar.retainSelection = true;
+    tabBar.mode = 'unfocused';
+    tabBar._updateTabs();
```
in general, we try to avoid directly calling `_update<Name>` methods directly as this doesn't simulate the component update cycle - instead we use `testRenderer.forceAllUpdates()`. You can see it used in the tests here for example https://github.com/rdkcentral/Lightning-UI-Components/blob/develop/packages/%40lightningjs/ui-components/src/components/Button/Button.test.js

benmunson committed:
address review comments

- changed test to use testRenderer update instead of calling internal TabBar method

svc-lightning-ui-components commented:
Test Execution Failed.

archaica commented:
'Text execution failed' is odd; it succeeded locally

svc-lightning-ui-components commented:
Test Execution Failed.

archaica committed:
Merge branch 'develop' into add-retain-selection

svc-lightning-ui-components commented:
Test Execution Passed.
